src/AppsCompiler.ts

```diff
@@ -87,6 +89,28 @@ export class AppsCompiler {
         return fs.promises.readFile(await packager.zipItUp());
     }
 
+    private validateAppPermissionsSchema(permissions: Array<IPermission>): void {
+        const examplePermissions = [{ name: 'user.read' }, { name: 'upload.write' }];
+        const error = new Error('Permissions declared in the app.json doesn\'t match the schema. '
+            + `It shoud be an peemissions array. e.g. ${ JSON.stringify(examplePermissions) }`);
+
+        if (!permissions) {
+            return;
+        }
+
+        if (!Array.isArray(permissions)) {
+            throw error;
+        }
+
+        if (permissions.length) {
```

Comment:
This `if` can be dropped, I think. As long as `permissions` is an array, there is no problem in calling the `.forEach` method on an empty array :)

Reply:
I think it is alright now